### PR TITLE
Group transactions by nonce

### LINKED ISSUE #5544: Group transactions by nonce
When speeding up or attempting to cancel a transaction, rather than display a new entry in the transaction history, we should group transactions as one and include the update (speed up / cancel) in the Activity log.

@cjeria to link to invision mocks, heads up @alextsg

### LINKED ISSUE #5732: "Speed up" button should ignore cancellation attempts
Right now the "speed up" shows on the most recent tx with lowest nonce.

If that most recent tx is a cancellation attempt, the "speed up" button shows on the cancellation attempt.

So, if I submit a tx, it takes a long time, I try to cancel, then I decide I should just speed up the original tx, I'm unable to do so since the "Speed up" button is now on the cancellation attempt and not on the original tx.

If a user tries to cancel, we should probably show the "Speed up" button on both the original transaction AND the cancellation attemp

inspired by [this comment](https://github.com/MetaMask/metamask-extension/issues/1410#issuecomment-437717020)

@alextsg

### LINKED ISSUE #5809: Show "Smart contract interaction" instead of "Unknown Function" in transaction history
They mean the same thing but one is way scarier! Even if we don't know the function name, the label on the home screen / transaction history can be way more helpful than it is now. Should be very simple to change. 

Relates to #5808

### PR DESCRIPTION
**Pull request by alextsg**
Fixes #5544 
Fixes #5732 
Fixes #5809

image
image
image

### CODE REVIEW

**Comment by alextsg** [comment]
@bdresser np, that won't be a huge change if we decide to do that.

**Comment by bdresser** [comment]
Notes from kyo sync:
* Current Etherscan button pops open the most recent tx – how do we link to Etherscan for the previous transactions? We can use the tiny little up arrow, could even link the whole line.
* Remove the transaction counter

**Comment by cjeria** [comment]
@alextsg @bdresser Here's an option for repositioning the attempts counter outside of the pill and adding a bit more copy for context. Thoughts?

image

**Comment by alextsg** [comment]
@cjeria I'm not too sure about the "No attempts made" option because we'd have a clickable link within a clickable container, and if that can be a confusing interaction for users. Also, I kind of see Cancel being grouped together with Speed Up in terms of functionality. Would it make sense to keep them together as they were before?

**Comment by alextsg** [comment]
image
Updated to remove the transaction counter and link each activity in the activity log to Etherscan.

**Comment by cjeria** [comment]
@alextsg 

> I'm not too sure about the "No attempts made" option because we'd have a clickable link within a clickable container

I agree. This was an attempt at surfacing the `speed up` button to address some complaints we've  recently received from support tickets that some users don't see or know that they can speed up the transaction. Let's keep it as you have it for now and we'll iterate on this some more. Maybe @bahnju can take a pass at some other options for surfacing up the `speed up` label per tx?

Re: the clickable activity link to etherscan - is the blue highlighted activity in your screenshot a hover state? If so, I'd suggest we make the hover state solid black on hover and use a lighter grey (same color as activity icons #9B9B9B) for the default state of all activity lines and `cursor:pointer` to indicate it's also a link.

Also, when did the details container bg color is get so dark? I have it at #FAFBFC in the designs.

image

Sorry for the nitpicks!

**Comment by alextsg** [comment]
@cjeria Yeah the blue was a hover state, and sure I'll make those color changes. I welcome the nitpicks!

File: ui/app/components/transaction-activity-log/transaction-activity-log-icon/transaction-activity-log-icon.component.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export default class TransactionActivityLogIcon extends PureComponent {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    className: PropTypes.string,
+    eventKey: PropTypes.oneOf([
```
Comment (whymarrh) [inline ui/app/components/transaction-activity-log/transaction-activity-log-icon/transaction-activity-log-icon.component.js]: Can we use the keys from `imageHash` here to keep them in sync?

File: ui/app/components/transaction-list-item/transaction-list-item.component.js [hunk 2]
```diff
@@ -51,36 +53,47 @@ export default class TransactionListItem extends PureComponent {
...
   }
 
-  handleRetry = () => {
+  /**
+   * @name handleRetry
+   * @description Resubmits a transaction. Retrying a transaction within a list of transactions with
+   * the same nonce requires keeping the original value while increasing the gas price of the latest
+   * transaction.
+   */
+  handleRetry = id => {
```
Comment (whymarrh) [inline ui/app/components/transaction-list-item/transaction-list-item.component.js]: If we're gonna a JSDoc here, we should document the `id` param too

File: ui/app/selectors/transactions.js [hunk 3]
```diff
@@ -36,23 +68,154 @@ export const transactionsSelector = createSelector(
...
+ *   initialTransaction: {},
+ *   primaryTransaction: {},
+ *   hasRetried: false,
+ *   hasCancelled: false,
+ * }
+ */
+export const nonceSortedTransactionsSelector = createSelector(
   transactionsSelector,
+  (transactions = []) => {
+    console.log('SELECTOR', transactions)
```
Comment (whymarrh) [inline ui/app/selectors/transactions.js]: Do we need this log here?

**Review by whymarrh — CHANGES_REQUESTED** [review]
A few small nitpicks. Some general feedback:

- It might be good to document the new functions added to `ui/app/selectors/transactions.js` from now while it's still fresh
- I skipped reading the CSS

**Comment by alextsg** [comment]
Updated PR

**Review by whymarrh — APPROVED** [review]

**Comment by whymarrh** [comment]
LGTM

**Pull request merged**